assets/quantity-popover.js

```diff
@@ -0,0 +1,47 @@
+class QuantityPopover extends HTMLElement {
+  constructor() {
+    super();
+    this.infoButton = this.querySelector('.variant-item__quantity-info.medium-hide');
+    this.popoverInfo = this.querySelector('.quantity-popover__info');
+    this.closeButton = this.querySelector('.button-close');
+
+    this.infoButton.addEventListener('click', this.togglePopover.bind(this));
+    this.closeButton.addEventListener('click', this.closePopover.bind(this));
+    this.infoButton.addEventListener('focusout', this.closePopover.bind(this));
+    this.popoverInfo.addEventListener('focusout', this.closePopover.bind(this));
+    this.addEventListener('mouseenter', this.toggleAccessibility.bind(this));
+    this.addEventListener('mouseleave', this.toggleAccessibility.bind(this));
+  }
+
+  togglePopover(event) {
+    event.preventDefault();
+
+    this.infoButton.setAttribute(
+      'aria-expanded',
+      (this.infoButton.getAttribute('aria-expanded') === 'false').toString()
+    );
+
+    if (this.infoButton.getAttribute('aria-expanded') === 'true') {
+      this.closeButton.focus();
+    }
+    this.toggleAccessibility(event);
+  }
+
+  toggleAccessibility(event) {
```

Comment:
Does this need to be its own method? Would it be simpler to include this in the method above?

``` Js
if (this.infoButton.getAttribute('aria-expanded') === 'true') {
  this.closeButton.focus();
}

this.popoverInfo.toggleAttribute('hidden');
```

Reply:
Some event listeners only need this portion, that is why I made it its own method https://screenshot.click/16-27-3flx3-few09.png